runtime/src/account_rent_state.rs

```diff
@@ -0,0 +1,135 @@
+use {
+    enum_iterator::IntoEnumIterator,
+    log::*,
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        rent::Rent,
+        transaction::{Result, TransactionError},
+        transaction_context::TransactionContext,
+    },
+};
+
+#[derive(Debug, PartialEq, IntoEnumIterator)]
+pub(crate) enum RentState {
+    Uninitialized, // account.lamports == 0
```

Comment:
nit: i like `Defaulted` as in bankrupt. ;)

Reply:
Hehe, I like that cleverness too :) However, I'm going to leave this as-is for now. We could `impl Default` for RentState in the future, though, if it seems like `RentState::default()` would be more useful or clear.